approval_polls/views.py

```diff
@@ -522,3 +522,44 @@ def post(self, request, *args, **kwargs):
             return HttpResponseRedirect(
                 reverse('approval_polls:embed_instructions', args=(p.id,))
             )
+
+
+class EditView(generic.View):
+
+    @method_decorator(login_required)
+    def get(self, request, *args, **kwargs):
+
+        # TODO handle polls that don't exist.
+        poll = Poll.objects.get(id=kwargs['poll_id'])
+        choices = Choice.objects.filter(poll=kwargs['poll_id'])
+        return render(request, 'approval_polls/edit.html', {
+            'poll': poll,
+            'choices': choices,
+            'closedatetime': poll.close_date.strftime("%Y/%m/%d %H:%M") if poll.close_date else ""
+        })
+
+    @method_decorator(login_required)
+    def post(self, request, *args, **kwargs):
+        poll = Poll.objects.get(id=kwargs['poll_id'])
+        closedatetime = request.POST['close-datetime']
+        try:
+            closedatetime = datetime.datetime.strptime(
+                closedatetime,
+                '%Y/%m/%d %H:%M'
+            )
+            current_datetime = timezone.localtime(timezone.now())
+            current_tzinfo = current_datetime.tzinfo
+            closedatetime = closedatetime.replace(
+                tzinfo=current_tzinfo
+            )
```

Comment:
I found the problem was with injecting the closedatetime into the template, not adding it back to the database. It was being converted to a string while still in UTC. Converting it to localtime before converting it makes the time stay the same between edits for me.

I'm assuming that we want the datetime dialog to work in the server's local time, since that's what the equivalent field on the create page is doing.